src/client/utils.ts

```diff
@@ -77,13 +88,14 @@ export const getSteps = async (
           `Pulled ${steps.length} steps from QStash` +
           ` and filtered them to ${filteredSteps.length} using messageId.`,
       });
-      return filteredSteps;
+      return { steps: filteredSteps, workflowRunEnded: false };
     }
   } catch (error) {
-    await debug?.log("ERROR", "ERROR", {
-      message: "failed while fetching steps.",
+    await debug?.log("WARN", "ENDPOINT_START", {
+      message:
+        "Couldn't fetch workflow run steps. This can happen if the workflow run succesfully ends before some callback is executed.",
       error: error,
     });
-    throw new WorkflowError(`Failed while pulling steps. ${error}`);
```

Comment:
I return this in all cases where this function throws. Very far from ideal. We need to narrow it down.